### ISSUE TITLE
true failure with threads::shared

### ISSUE THREAD

**Issue by sbingner** [rurban/Cpanel-JSON-XS#166]
I have the below behavior with Cpanel::JSON::XS v 4.19 - the issue is that it returns *every* boolean value as false.

```perl
use Cpanel::JSON::XS qw(encode_json decode_json);
use Data::Dumper;
use threads;
use threads::shared;
use strict;

my $json : shared = shared_clone({'enabled' => Cpanel::JSON::XS::true});

print encode_json($json) . "\n" . Data::Dumper->Dump([$json]) . "\n\n";
```
output:
```
{"enabled":false}
$VAR1 = {
          'enabled' => bless( do{\(my $o = 1)}, 'JSON::PP::Boolean' )
        };
```

But when I try with stock JSON:
```perl
use JSON::XS qw(encode_json decode_json);
use Data::Dumper;
use threads;
use threads::shared;
use strict;

my $json : shared = shared_clone({'enabled' => JSON::XS::true});

print encode_json($json) . "\n" . Data::Dumper->Dump([$json]) . "\n\n";
```
output:
```
{"enabled":true}
$VAR1 = {
          'enabled' => bless( do{\(my $o = 1)}, 'JSON::PP::Boolean' )
        };
```

**Comment by rurban** [comment]
Added a testcase, thanks

**Comment by rurban** [comment]
Fixed in master, will be in the upcoming 4.20
Thanks

**Comment by rurban** [comment]
Note that JSON::PP from 2.9x to 4.0 are broken also. You need to upgrade.
